packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2361,14 +2362,32 @@ abstract class RenderBox extends RenderObject {
       _cachedBaselines?.clear();
       _cachedIntrinsicDimensions?.clear();
       _cachedDryLayoutSizes?.clear();
-      if (parent is RenderObject) {
-        markParentNeedsLayout();
-        return;
-      }
+      _notifyParentIfDirty = true;
+    }
+  }
+
+  @override
+  void markNeedsLayout() {
+    _clearCachedData();
+    if (_notifyParentIfDirty && parent is RenderObject) {
+      markParentNeedsLayout();
+      _notifyParentIfDirty = false;
+      return;
     }
     super.markNeedsLayout();
   }
 
+  @override
+  void layout(Constraints constraints, {bool parentUsesSize = false}) {
+    if (hasSize && constraints != this.constraints &&
+        _cachedBaselines != null && _cachedBaselines!.isNotEmpty) {
+      // The cached baselines data may need update if the constraints change.
+      _cachedBaselines?.clear();
+      _notifyParentIfDirty = true;
```

Comment:
I don't think we need to keep track of the dirtiness here. Baseline is only valid after layout has been called, so if the parent is interested in that metric, they would call it after layout and with that re-establish the dependency.

Reply:
You are right and fixed.